### PR TITLE
update initializer.Normal API

### PR DESCRIPTION
**Pull request by windstamp**
### PR types

Function optimization

### PR changes

APIs

### Describe

update initializer API, including: NormalInitializer, TruncatedNormalInitializer, UniformInitializer, XavierNormalInitializer, XavierUniformInitializer, NumpyArrayInitializer.

这次修改涉及的场景比较麻烦，好像在迁移之前就已经存在别名问题导致的文档问题 http://newicafe.baidu.com/v5/issue/DLTP-13276/show。 如果 XXX、XXXInitializer 接口都需要暴露，则必须都定义，不能再使用 XXX = XXXInitializer 的写法。前面的链接里面有讨论结论。比如，对于 Normal 和 NormalInitializer API，之前使用 Normal = NormalInitializer 别名方式，现在是重新定义了一个 类 Normal，继承自 NormalInitializer。

另外：
1. XavierNormalInitializer 和 XavierUniformInitializer 由于是新增 API，应该是放在 paddle.nn 目录实现的，但是鉴于整体代码结构，我这里还是放在 paddle.fluid 实现，采用别名方式映射到 paddle.nn 目录中。
2. XavierNormalInitializer 和 XavierUniformInitializer 不使用参数 gain，而继续使用参数 seed。这是之前调研时对于参数 gain 的意思有误解，它不是 seed 的意思，是系数（factor）的意思，所以这里还是保持和原来的实现一致，不改变 API 的具体实现。tensorflow 也是参数 seed，不是 gain，pytorch 是 gain。

我这里先提供一个初步的版本，麻烦看看哪里需要修改，请指出。

迁移到另一个 [PR#27744](https://github.com/PaddlePaddle/Paddle/pull/27744) 了，关闭此 PR。

### CODE REVIEW

**Comment by windstamp** [comment]
NumpyArrayInitializer doc preview.

image

**Comment by windstamp** [comment]
XavierNormalInitializer doc preview.

image

**Comment by windstamp** [comment]
XavierUniformInitializer doc preview.

image

**Comment by windstamp** [comment]
这次修改涉及的场景比较麻烦，好像在迁移之前就已经存在别名问题导致的文档问题 http://newicafe.baidu.com/v5/issue/DLTP-13276/show。 如果 XXX、XXXInitializer 接口都需要暴露，则必须都定义，不能再使用 XXX = XXXInitializer 的写法。前面的链接里面有讨论结论。

我这里先提供一个初步的版本，麻烦看看哪里需要修改，请指出。

**Comment by windstamp** [comment]
NormalInitializer doc preview.

image

**Comment by windstamp** [comment]
TruncatedNormalInitializer doc preview.

image

**Comment by windstamp** [comment]
UniformInitializer doc preview.

image
